web_poet/requests.py

```diff
@@ -0,0 +1,132 @@
+        However, the underlying downloader assigned in
+        ``web_poet.request_backend_var`` might change that, depending on
+        how the framework using **web-poet** implements it.
+    """
+
+    logger.info(f"Requesting page: {request}")
+
+    try:
+        request_backend = request_backend_var.get()
+    except LookupError:
+        raise RequestBackendError(
+            "Additional requests are used inside the Page Object but the "
+            "current framework has not set any HttpRequest Backend via "
+            "'web_poet.request_backend_var'"
+        )
+
+    response_data: HttpResponse = await request_backend(request)
+    return response_data
+
+
+class HttpClient:
+    """A convenient client to easily execute requests.
+
+    By default, it uses the request implementation assigned in the
+    ``web_poet.request_backend_var`` which is a :mod:`contextvars` instance to
+    download the actual requests. However, it can easily be overridable by
+    providing an optional ``request_downloader`` callable.
+
+    Providing the request implementation by dependency injection would be a good
+    alternative solution when you want to avoid setting up :mod:`contextvars`
+    like ``web_poet.request_backend_var``.
+
+    In any case, this doesn't contain any implementation about how to execute
+    any requests fed into it. When setting that up, make sure that the downloader
+    implementation returns a :class:`~.HttpResponse` instance.
+    """
+
+    def __init__(self, request_downloader: Callable = None):
+        self.request_downloader = request_downloader or _perform_request
+
+    async def request(
+        self,
+        url: str,
+        *,
+        method: str = "GET",
+        headers: Optional[Headers] = None,
+        body: Optional[Body] = None,
+    ) -> HttpResponse:
+        """This is a shortcut for creating a :class:`HttpRequest` instance and executing
+        that request.
+
+        A :class:`~.HttpResponse` instance should then be returned.
+
+        .. warning::
+            By convention, the request implementation supplied optionally to
+            :class:`~.HttpClient` should return a :class:`~.HttpResponse` instance.
+            However, the underlying implementation supplied might change that,
+            depending on how the framework using **web-poet** implements it.
+        """
+        headers = headers or {}
+        body = body or b""
+        req = HttpRequest(url=url, method=method, headers=headers, body=body)
+        return await self.request_downloader(req)
+
+    async def get(self, url: str, *, headers: Optional[Headers] = None) -> HttpResponse:
+        """Similar to :meth:`~.HttpClient.request` but peforming a ``GET``
+        request.
+        """
+        return await self.request(url=url, method="GET", headers=headers)
+
+    async def post(
+        self,
+        url: str,
+        *,
+        headers: Optional[Headers] = None,
+        body: Optional[Body] = None,
```

Comment:
If I'm not mistaken, technically, even "bytes" annotation could be ok, because HttpRequestBody is a subclass of bytes. But it'd won't allow to show the intent here.